cmd/fleet/cron.go

```diff
@@ -190,32 +190,51 @@ func cronVulnerabilities(
 				appConfig = freshAppConfig
 			}
 
-			var vulnAutomationEnabled bool
+			var vulnAutomationEnabled string
 
-			// only one of the webhook or a jira integration can be enabled at a
+			// only one vuln automation (i.e. webhook or integration) can be enabled at a
 			// time, enforced when updating the appconfig.
 			if appConfig.WebhookSettings.VulnerabilitiesWebhook.Enable {
-				vulnAutomationEnabled = true
-			} else {
-				for _, intg := range appConfig.Integrations.Jira {
-					if intg.EnableSoftwareVulnerabilities {
-						vulnAutomationEnabled = true
-						break
+				vulnAutomationEnabled = "webhook"
+			}
+			// check for jira integrations
+			for _, j := range appConfig.Integrations.Jira {
+				if j.EnableSoftwareVulnerabilities {
+					if vulnAutomationEnabled != "" {
+						level.Error(logger).Log("err", "more than one automation enabled")
+						// TODO: additional error handling?
+					}
+					vulnAutomationEnabled = "jira"
+					break
+				}
+			}
+			// check for zendesk integrations
+			for _, z := range appConfig.Integrations.Zendesk {
+				if z.EnableSoftwareVulnerabilities {
+					if vulnAutomationEnabled != "" {
+						level.Error(logger).Log("err", "more than one automation enabled")
+						// TODO: additional error handling?
 					}
+					vulnAutomationEnabled = "zendesk"
+					break
 				}
 			}
+			level.Debug(logger).Log("vulnAutomationEnabled", vulnAutomationEnabled)
 
-			recentVulns := checkVulnerabilities(ctx, ds, logger, vulnPath, config, vulnAutomationEnabled)
-			if vulnAutomationEnabled && len(recentVulns) > 0 {
-				if appConfig.WebhookSettings.VulnerabilitiesWebhook.Enable {
+			recentVulns := checkVulnerabilities(ctx, ds, logger, vulnPath, config, (vulnAutomationEnabled != ""))
+			if len(recentVulns) > 0 || true {
+				switch vulnAutomationEnabled {
+				case "webhook":
 					// send recent vulnerabilities via webhook
 					if err := webhooks.TriggerVulnerabilitiesWebhook(ctx, ds, kitlog.With(logger, "webhook", "vulnerabilities"),
 						recentVulns, appConfig, time.Now()); err != nil {
 
 						level.Error(logger).Log("err", "triggering vulnerabilities webhook", "details", err)
 						sentry.CaptureException(err)
 					}
-				} else {
+
+				// TODO: what about jobs that get queued but are not run before reconfig automations?
```

Comment:
Good question. We might want to ignore those jobs and mark them as failed with an error like "automation not configured" potentially.